ts-apps/jl4-web/src/routes/+page.svelte

```diff
@@ -0,0 +1,247 @@
+<script lang="ts">
+  import { onMount } from "svelte";
+  let editorElement: HTMLDivElement;
+  import {
+  LadderFlow,
+    LirContext,
+    LirRegistry,
+    VizDeclLirSource,
+    type DeclLirNode,
+    type LirRootType,
+  } from "@repo/decision-logic-visualizer";
+  import {
+    type VisualizeDecisionLogicIRInfo,
+    type FunDecl,
+  } from "@repo/viz-expr";
+  import { type MessageTransports } from "vscode-languageclient";
+  import { type ConsoleLogger } from "monaco-languageclient/tools";
+
+  /**************************
+      Set up Lir
+  ****************************/
+
+  const registry = new LirRegistry();
+  const context = new LirContext();
+  const nodeInfo = { registry, context };
+
+  let vizDecl: undefined | FunDecl = $state(undefined);
+  let declLirNode: DeclLirNode | undefined = $derived(
+    vizDecl && VizDeclLirSource.toLir(nodeInfo, vizDecl)
+  );
+  let funName = $derived(
+    declLirNode && (declLirNode as DeclLirNode).getFunName(context)
+  );
+  $effect(() => {
+    if (declLirNode) {
+      registry.setRoot(context, "VizDecl" as LirRootType, declLirNode);
+    }
+  });
+
+  // /**************************
+  //       Monadco
+  // ****************************/
+
+  onMount(async () => {
+    const monaco = await import("@codingame/monaco-vscode-editor-api");
+    const { initServices } = await import(
+      "monaco-languageclient/vscode/services"
+    );
+    const { LogLevel } = await import("@codingame/monaco-vscode-api");
+    const { CloseAction, ErrorAction } = await import(
+      "vscode-languageclient/browser.js"
+    );
+    const { MonacoLanguageClient } = await import("monaco-languageclient");
+    const { WebSocketMessageReader, WebSocketMessageWriter, toSocket } =
+      await import("vscode-ws-jsonrpc");
+    const { configureDefaultWorkerFactory } = await import(
+      "monaco-editor-wrapper/workers/workerLoaders"
+    );
+    const { ConsoleLogger } = await import("monaco-languageclient/tools");
+    const { Schema } = await import("effect");
```

Comment:
How come `const { Schema } = await import("effect");` needs to be imported like this?

Reply:
The typescript error doesn't go away if we don't import it that way but I can try whether it works (while ignoring the type error) and change it.

Reply:
I can experiment with that, I just wanted to try to understand the intent so I don't inadvertently break something

Reply:
@ym-han btw if you have any idea of how to solve this type error or make it simpler s.t. it can't show a type error, that would be great.

Reply:
> @ym-han btw if you have any idea of how to solve this type error or make it simpler s.t. it can't show a type error, that would be great. (because I personally cannot make sense of the error)

I'll experiment with it, yeah. But it would helpful if you could explain the intent behind those await imports in the first place so that I don't accidentally break something or waste too much time going down the wrong path. I don't have that much frontend experience.

Reply:
Yep!